sdk/freeport/freeport.go

```diff
@@ -71,6 +71,17 @@ var (
 // initialize is used to initialize freeport.
 func initialize() {
 	var err error
+
+	blockSize = 1500
+	limit, err := systemLimit()
+	if err != nil {
+		panic("freeport: error getting system limit: " + err.Error())
+	}
+	if limit > 0 && limit < blockSize {
+		logf("INFO", "blockSize %d too big for system limit %d. Adjusting...", blockSize, limit)
+		blockSize = limit - 3
+	}
+
```

Comment:
IS this solving the darwin annoying warning in tests? If so cool :) I wasn't sure if it's relevant to this PR or not?

Reply:
I don't think so. But it solves the case where your system limits are lower than the lower bound of freeport, which makes its own test fail instantly on OSX.